applications/FluidDynamicsApplication/custom_strategies/schemes/residualbased_simple_steady_scheme.h

```diff
@@ -104,6 +105,14 @@ class ResidualBasedSimpleSteadyScheme : public Scheme<TSparseSpace, TDenseSpace>
     mPressureRelaxationFactor = factor;
   }
 
+  void Initialize(ModelPart& rModelPart) override
+  {
+      BaseType::Initialize(rModelPart);
+
+      // This is required only for steady state adjoints.
+      VariableUtils().SetNonHistoricalVariableToZero(RELAXED_ACCELERATION, rModelPart.Nodes());
```

Comment:
This means we are wasting memory if this scheme is used in a non-adjoint simulation right? We can leave this as a temporary solution, but in my opinion we should introduce something to distinguish between adjoint and non-adjoint simulations. Maybe something as easy as `IS_ADJOINT_SIMULATION` in the `ProcessInfo` can do the job.

Reply:
I think this is how it is done in structural

right @armingeiser @MFusseder ?

Reply:
Not sure if i got you right, but in StructMech, the historical adjoint variables are only added by the adjoint solver (py)

Reply:
@armingeiser What you are saying is perfectly ok I think. 

The line which is mentioned here is done like that because of the following.
In Fluid/RANS we have steady state adjoints as well as transient adjoints. But we use the same primal/adjoint elements in both cases. So in the steady state case we make the above `RELAXED_ACCELERATION` variable to zero to remove transient contributions from adjoints. 

Otherwise, we have to create two elements for steady and transient adjoints. The time scheme is hardcored to the adjoint elements, so there is no possiblity to change it in the scheme level I think. The element interface does not support time schemes in the adjoint case.

Reply:
I meant this:
https://github.com/KratosMultiphysics/Kratos/blob/8420897552a2aa87a9b071a0fab433339318dd89/applications/StructuralMechanicsApplication/structural_mechanics_application_variables.h#L327

Reply:
Hmm.... Then I can take the core `ResidualBasedAdjointStaticScheme` and make a new adjoint scheme in fluidApp so we do this only for the adjoint runs, so we can keep the primal runs unchanged? Atm, I am directly using the core `ResidualBasedAdjointStaticScheme`

Reply:
Removed to `adjoint_monolithic_solver.py`. Now we don't have additional cost in the primal solve if adjoints are not used.